apps/docs/src/guide/contracts/deploying-contracts.md

```diff
@@ -27,11 +27,23 @@ Load the contract bytecode and JSON ABI, generated from the Sway source, into th
 
 ## 4. Deploying the Contract
 
-Initialize a [`ContractFactory`](../../api/Contract/ContractFactory.md) with the bytecode, ABI, and wallet. Deploy the contract and use its methods.
+To deploy the contract we can instantiate the [`ContractFactory`](../../api/Contract/ContractFactory.md) with the bytecode, ABI, and wallet. The we can call `deployContract` method. This method returns a promise that will resolve to a `Contract` instance, ready to be used to interact with the deployed smart contract.
 
 <<< @/../../docs-snippets/src/guide/contracts/deploying-contracts.test.ts#contract-setup-3{ts:line-numbers}
 
-## 5. Executing a Contract Call
+The `deployContract` method will only resolve after the deploy contract transaction has been processed and the smart contract is deployed on the blockchain. As a result, this call can take several seconds to resolve and will block any code execution that follows.
+
+If this situation is inappropriate for your dApp, you can use the `deployContractAsync` method.
+
+## 5. Deploying the Contract Asynchronously
+
+In some cases, you may not want to wait for the deploy contract transaction to finish processing. In these instances, you can use `deployContractAsync`. This method resolves as soon as the transaction to deploy the contract is submitted and returns two objects: a `TransactionResponse` and a `Contract` instance.
```

Comment:
I'd probably make the point about the `waitForResult` here as that seems like the most important point here, and the key distinction between the 2 methods.

Reply:
@danielbate The docs are slightly different now https://github.com/FuelLabs/fuels-ts/pull/2597/commits/7c800e1786393467ffb92433a7a331d6ca38d4cb